src/platform/intel/cavs/include/cavs/lib/pm_memory.h

```diff
@@ -154,4 +154,27 @@ void cavs_pm_memory_lp_sram_power_gate(void *ptr, uint32_t size, bool enabled);
 
 #endif /* CONFIG_LP_SRAM */
 
+#if CONFIG_L1_DRAM
+
+static inline void cavs_pm_memory_l1_dram_banks_power_gate(
+	uint32_t start_bank_id, uint32_t ending_bank_id, bool enable)
+{
+	uint32_t mask = MASK(start_bank_id, ending_bank_id);
+	uint32_t pgctl = io_reg_read(L1_MEM_DRAM_PGCTL);
+	uint32_t pgists = io_reg_read(L1_MEM_DRAM_PGISTS);
+	uint32_t expected = enable ? pgists & mask : pgists | ~mask;
+
+	if (enable)
+		pgctl &= mask;
+	else
+		pgctl |= ~mask;
+
+	io_reg_write(L1_MEM_DRAM_PGCTL, pgctl);
+
+	while ((io_reg_read(LSPGISTS) & mask) != expected)
```

Comment:
We will need a timeout here as we dont want to spin forever if the host or other FW code mis-configures HW.

Reply:
Oh yes, good point.

Reply:
@lrgirdwo I have added timeout handling for all functions that poll mem registers.